src/hb-common.cc

```diff
@@ -1039,6 +1043,21 @@ hb_variation_from_string (const char *str, int len,
   return false;
 }
 
+#if !defined(HARFBUZZ_NO_SETLOCALE) && defined(HAVE_NEWLOCALE) && defined(HAVE_USELOCALE)
+static locale_t
+get_C_locale (void)
+{
+  static locale_t C_locale = NULL;
+
+  if (!C_locale)
+    C_locale = newlocale (LC_ALL_MASK, "C", NULL);
```

Comment:
Assuming `locale_t` is of pointer size, we would use cmpexch for this initialization. And we have `hb_lazy_loader_t` to do the cruft. Check `hb-ft.cc` for how to use it for a type we think is a pointer underneath:
```
tatic struct hb_ft_library_lazy_loader_t : hb_lazy_loader_t<hb_remove_pointer<FT_Library>,
                                                             hb_ft_library_lazy_loader_t>
{
...
}...;
```

Reply:
will have a look, thanks

Reply:
I did not find any guarantees in POSIX about the size or or characteristics of locale_t, but at least in glibc, its a pointer to a struct.

i gave the cmpexch approach a try.